spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/MqttPahoMessageDrivenChannelAdapter.java

```diff
@@ -179,46 +180,58 @@ protected void doStart() {
 	}
 
 	@SuppressWarnings("deprecation")
-	private synchronized void connect() throws MqttException {
-		MqttConnectOptions connectionOptions = this.clientFactory.getConnectionOptions();
-		var clientManager = getClientManager();
-		if (clientManager == null) {
-			Assert.state(getUrl() != null || connectionOptions.getServerURIs() != null,
-					"If no 'url' provided, connectionOptions.getServerURIs() must not be null");
-			this.client = this.clientFactory.getAsyncClientInstance(getUrl(), getClientId());
-			this.client.setCallback(this);
-			this.client.connect(connectionOptions).waitForCompletion(getCompletionTimeout());
-			this.client.setManualAcks(isManualAcks());
+	private void connect() throws MqttException {
+		this.lock.lock();
```

Comment:
Do you really think that connection monitor and topics monitor should be same instance?
I believe there was a reason to have them separately.

Reply:
Good point.  
Looked like the parent and child locks always guard the entire method calls, e.g. like `this` monitor.
But after our remark I found one exception ;) So I re-did the conversion for the mqtt classes from scratch.